### ISSUE TITLE
It is recommended that all images for the theme Twenty Twenty-Five be in avif image format.

### ISSUE THREAD

**Issue by phanduynam** [WordPress/twentytwentyfive#383]
Currently, all images in the Twenty Twenty-Four theme are in webp image format. I would like all images in the upcoming Twenty Twenty-Five theme to use avif image format. Avif image format will help improve image performance and quality on Wordpress much more than webp.

**Comment by JosVelasco** [comment]
Since WordPress has supported AVIF since 6.5, and one of the theme's requirements is 6.6+ (it will be 6.7 at the time of release), I'd love to see this as well to promote AVIF.

**Comment by JosVelasco** [comment]
WebP folder: 5.7 MB.
AVIF folder: 3.5 MB.

I used ImageMagick's default conversion mode, so some texture grain details are blurred, especially on some faces. However, there is a big difference in size.

Take a look:
https://drive.google.com/drive/folders/1KfUsoRLRtboHl8N7P0gthQjz45ePQr7q?usp=sharing

**Comment by chrisdavidmiles** [comment]
Some questions were asked in slack #core-themes slack about avif/webp compatibility with the major WordPress hosts so I'm adding comments here about Bluehost where I work. At Bluehost when a customer signs up for WordPress hosting we put them on the latest version of WordPress, latest available version of PHP so that new sites get built with the latest software by default. As of now avif/webp is supported at Bluehost for any new signups (more specifically the imagemagick and php extensions available support core's implementation of avif/webp). There are some existing customers on older hardware that don't have support yet, but Bluehost is actively upgrading those servers as well. 💙

**Comment by JosVelasco** [comment]
To test Avif support on TT5, I forked the theme and edited the pattern Banner with the book description to include an Avif file.

I'm not sure if this is the best approach, but I think we could follow these instructions:

- Install and activate Gutenberg
- Install the forked theme https://github.com/JosVelasco/twentytwentyfive
- Add a new Page or post
- Insert the pattern Banner with book description
- Click Publish
- Confirm whether the suggestion to import external media appears.
- Confirm the image was successfully imported and srcset is used in the front-end.

**Comment by Faizan-Nabi-Web** [comment]
I tested Avif support on TT5 using Rocket.net and can confirm it works without any issues on the latest wp core (6.6.2).

**Comment by carolinan** [comment]
During the weekly meeting on Slack, it was brought up that server support may not be an issue since the images
are included in the theme and placed in patterns using `get_template_directory_uri()`

But WordPress has a feature where users are asked if they want to save the images to the Media Library. 
And this is the feature that needs to be looked into: If it can not be turned off if the image format is not supported, then using AVIF may not be possible.

**Comment by JosVelasco** [comment]
For my testing, I disabled AVIF support by filtering the list of allowed mime types for uploads, and after trying to upload the external images with the Upload button, I got the error: 

**Upload failed, try again.** 

```
function disable_avif_mime_types( $mime_types ) {
    // Unset AVIF image types
    if ( isset( $mime_types['avif'] ) ) {
        unset( $mime_types['avif'] );
    }
    return $mime_types;
}
add_filter( 'upload_mimes', 'disable_avif_mime_types' );
```

This error is not very informative. Still, this function is not a blocker with servers that don't support AVIF because it is possible to publish the post without uploading or replacing the images.

Other things that could make the upload process fail include firewalls. But I think this is more for the Hosting companies to fix. What else should we check for? @carolinan 

More information is available in [Prepublish: suggest uploading external images #46014](https://github.com/WordPress/gutenberg/pull/46014).

**Comment by aaroncampbell** [comment]
Our product team at A2 Hosting tested this on the default setups on all our product lines. Every one displays the AVIF using srcset (from uploads) as expected.

**Comment by carolinan** [comment]
I suggest postponing this until after the 6.7 release.
The bundled themes can be updated at any time outside a major or minor WordPress release.

- By postponing it, we can ensure the format is not used in the theme before any initial bugs are solved.
- The downside is that the theme can not be used as a way to help test the format.

**Comment by JosVelasco** [comment]
I wish there were a particular beta version for these tests that are not too technical and require manual testing.

Can we include them as an experiment somehow?

Another way could be establishing conditions, such as if several Hosts support it, we try it. I saw the willingness of Hosts to test.

**Comment by carolinan** [comment]
The images in the theme does not need to be avif, to test avif all you need is any image of this format.

**Comment by felixarntz** [comment]
I just tested using AVIF in TT5, quickly replaced the WebP images for a pattern with AVIF in my local theme's checkout (for testing I used the two flower images from the `banner-with-description-and-images-grid.php` pattern).

I used WordPress 6.7 Beta 1 on a PHP 7.4 host, which therefore for sure does not support processing AVIF (AVIF is only supported in certain Imagick or GD versions that require PHP 8+).

On that test site, when I add the pattern to a page, it works fine. Of course the AVIF image is displayed correctly, as it's supported by the browser. However, when I click the button to upload the image to the Media library, it fails with an error: `undefined: Sorry, this file type is not supported here.` See screenshot:

![Image](https://github.com/user-attachments/assets/9f0726c6-637d-46f8-89c7-6fbb32eebad0)

In principle, uploading the AVIF in this situation shouldn't be a real problem because it can still be displayed. The only thing that wouldn't work is that WordPress couldn't generate the sub-sizes for that image. In this particular scenario I'd consider that an edge-case though because:
* I assume very few sites would reuse the "demo" images shipped with a theme anyway.
* If they wanted to, and the server didn't support AVIF processing, uploading them to the Media Library would not be any worse from a performance perspective than loading the image directly from the theme: In neither of those situations, the sub-sizes would exist.
* The only situation where the missing sub-sizes would actually lead to a bug, i.e. broken UI, would be if someone's site was using a custom hard-crop image size and then adjusted the image in the pattern to use that other aspect ratio. Here, the sub-size missing would mean the original image would appear distorted.

The severity of this would be extremely low because all of those 3 conditions with low probability would need to be met. It would probably affect less than 1% of sites. So it may be worth considering to allow AVIF upload in such a situation and simply skip generation the sub-sizes.

That's a concern for the Core Media team though and would need to be discussed separately. As it stands today, WordPress Core prevents to upload an AVIF in this scenario, and because of that the user experience is more severely broken (again, more than it needs to IMO, but here we are). Based on this, I agree that it's too early to include AVIF images in the default theme. It would be a great performance signal, but it cannot come at the cost of other UX aspects that are as widespread as the lack of AVIF processing support of Imagick and GD is today.

**Comment by phanduynam** [comment]
Does anyone know what minimum php version WordPress 6.7 will officially support?

**Comment by JosVelasco** [comment]
@phanduynam According to the [Roadmap to 6.7](https://make.wordpress.org/core/2024/09/03/roadmap-to-6-7/):

### Continuing to improve PHP 8.x support
To continue improving support for PHP 8.x, code specific to prior PHP versions that are no longer supported [have been removed](https://core.trac.wordpress.org/ticket/61574).

**Comment by desrosj** [comment]
There will be no change to the minimum version of PHP required to run WordPress in 6.7. It will [remain at 7.2.24 as recently adjusted in WordPress 6.6](https://make.wordpress.org/core/2024/04/08/dropping-support-for-php-7-1/).

I talked this through with @felixarntz today at length, and his comment above summarizes the discussion well. I'd like to confirm that we've optimized the WebP images as much as possible. There is an `imagemin` script that runs for all image files in WordPress Core and bundled default themes, but that does not seem to support WebP format.

I'm going to second the suggestion to close this as not planned.
